davidpak/SmartNote
# Finished generate RPC

## What type of PR is this? (check all applicable)


## Description

Finished the `generate` RPC and made updates to the configuration file.

## QA Instructions, Screenshots, Recordings

The `generate` RPC behaves as described in `GENERATE.md` and `GENERATION.md`, but requires some setup to be used correctly in `config.json`. The additional fields are documented in `CONFIGURATION.md`.

Here is a brief overview of how the configuration should be updated:

* `summarizer`: Set to the location of the `summarizer.py` script. By default, this is `scripts/src/summarizer.py`.
* `env`: Set to the location of the `.env` file for the summarizer, which gives the OpenAI key. By default, this is `private/.env`. The file should contain the single line: `OPENAI_API_KEY=<key>`.
* `python`: Set to the location of the Python interpreter to use. Must be a Python 3 interpreter. By default this is `python3`, meaning the path will be searched for the executable. This will likely need to change. The interpreter should be in an environment in which all the dependencies for `summarize.py` are installed. This is usually an Annoconda environment.
* `debug`: Enable/disable debug mode. Debug mode skips the summarizer and instead always treats the resource located at `debugResource` as the generated summary.
* `debugResource`: The resource to treat as the generated summary, if in debug mode.

## Added/updated tests?
